tyrian/js/src/main/scala/tyrian/TyrianRoutedAppF.scala

```diff
@@ -0,0 +1,45 @@
+package tyrian
+
+import cats.effect.kernel.Async
+import cats.effect.kernel.Resource
+import org.scalajs.dom.Element
+import org.scalajs.dom.document
+import tyrian.runtime.TyrianRuntime
+
+import scala.scalajs.js.annotation._
+
+/** The TyrianApp trait can be extended to conveniently prompt you for all the methods needed for a Tyrian app, as well
+  * as providing a number of standard app launching methods.
+  */
+trait TyrianRoutedAppF[F[_]: Async, Msg, Model] extends TyrianAppF[F, Msg, Model]:
+
+  def hashResultToMessage: Navigation.Result => Msg
+  def hashChangeToMessage: Navigation.Result.HashChange => Msg
+
+  def setLocationHash(newHash: String): Cmd[F, Nothing] =
+    Navigation.setLocationHash(newHash)
+
+  def _init(flags: Map[String, String]): (Model, Cmd[F, Msg]) =
```

Comment:
Perhaps you might want to make all these `_`-prefixed methods private? Otherwise they'll leak into `SinglePage` and `MultiPage`, is that on purpose? 🤔